server/internal/client/client.go

```diff
@@ -185,10 +186,9 @@ type ConnHandle struct {
 // the other waits of the results of these jobs and writes them to the client.
 func SpawnConn(conn Conn) *ConnHandle {
 	c := &ConnHandle{
-		conn: conn,
-		// responses needs to have an extra slot for the sentinal value sent by
-		// Destroy
-		responses: make(chan interface{}, maxOutstandingRequests+2),
+		conn:      conn,
+		responses: make(chan interface{}, maxOutstandingRequests+1),
```

Comment:
Do we still need the extra slot?

Reply:
I changed it from +2 to +1 b/c I wasn't sure why it was +2 in the first place (rather than the +1 the comment explained). But now that I think about it may be because both Destroy and the health check goroutine could call destroy? In which case I can remove the +1 here.

Reply:
That was it